### PR TITLE
Code review for Reinder to have fun

### PR DESCRIPTION
**Pull request by Asanto32**

### CODE REVIEW

**Comment by ReinderVosDeWael** [comment]
> When everything is addressed we can merge and I make a few issues about the architecture so we can switch to smaller PRs that are easier to review (Unless @ReinderVosDeWael is opposed).

I'd recommend freezing development on this branch (after including Florian's minor suggestions) and then making branches that split off and merge into _this_ branch. We should treat the main branch as production code i.e. it doesn't go to main until it receives full approval, but the feature branches merging into `Adam_woking_v2` are evaluated based on only their completeness within their own scope. 

@vitoetc, @gkiar are there any real harms caused by delaying merging this to main? Code coverage of 35% is sufficient by itself for me to recommend not releasing this yet, but resolving that puts us in a bind where either we 1) write tests for the current architecture that have to be rewritten upon making the required architectural changes (i.e. doubling work), or 2) we have to fix the architectural issues before getting to testing (i.e. delaying release).

File: src/wristpy/ggir/calibration.py [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+        )
+        if not finished and (nh + i_h * n12h) >= accel_data.shape[0]:
+            finished = True
+            valid_calibration = False
+            warn(
+                f"Calibration not done with {min_hours + (i_h - 1) * 12} - "
+                f"{min_hours + i_h * 12} hours due to insufficient non-movement "
+                f"data available"
+            )
+            return OutputData(
```
[the remark this answers is not in the document]
Comment (Asanto32) [reply src/wristpy/ggir/calibration.py:110]: I'm not sure I understand this comment? The finished logic is only defined later when the iterative process is used (which will hopefully be removed in the second, improved, version of calibration)

File: src/wristpy/ggir/calibration.py [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+        and the scale and offset value from calibration.
+    """  # noqa: E501
+    accel_data = input_data.acceleration
+    time_data = input_data.time
+    sampling_rate = input_data.sampling_rate
+
+    nh = int(min_hours * 3600 * sampling_rate)
+    n12h = int(12 * 3600 * sampling_rate)
+
+    i_h = 0  # keep track of number of extra 12 hour blocks used
```
[the remark this answers is not in the document]
Comment (Asanto32) [reply src/wristpy/ggir/calibration.py:62]: Changed, but these will be removed when iterative cal processed is removed

File: src/wristpy/ggir/calibration.py [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+            Tolerance for stopping iteration. Default is 1e-10. Generally this should be left at this
+            value.
+
+    Returns:
+        Output: Output data class with the calibrated acceleration data, the start and end calibration error,
+        and the scale and offset value from calibration.
+    """  # noqa: E501
+    accel_data = input_data.acceleration
+    time_data = input_data.time
+    sampling_rate = input_data.sampling_rate
```
[the remark this answers is not in the document]
Comment (Asanto32) [reply src/wristpy/ggir/calibration.py:57]: probably not, but it shortens it...

File: src/wristpy/ggir/error_tests.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
+"""Compute error for ristpy vs GGIR."""
```
[the remark this answers is not in the document]
Comment (Asanto32) [reply src/wristpy/ggir/error_tests.py:1]: Removing this from repo, keeping local copy only

File: src/wristpy/ggir/error_tests_local.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    Returns:
+        A tuple containing the mean squared error for anglez, mean squared error for ENMO, and the median difference of anglez.
+    """  # noqa: E501
+    epoch1_data = compare_dataframes.compare_csv(
+        ggir_dataframe=ggir_data, wristpy_dataframe=wristpy_data
+    )
+
+    # extend non-wear flag to smooth out edges
+    epoch1_data = epoch1_data.with_columns(pl.col("time_epoch1").set_sorted())
+    NW_flag_rolling_mean = epoch1_data.group_by_dynamic(
```
[the remark this answers is not in the document]
Comment (Asanto32) [reply src/wristpy/ggir/error_tests_local.py:86]: Moved out of repo to local

**Comment by Asanto32** [comment]
TODO
Fix README before merging with main?

**Comment by gkiar** [comment]
@ReinderVosDeWael — I support your recommendation here. Please sketch out the concrete roadmap for development here with @Asanto32 and @frey-perez , then run it by @vitoetc and myself for a final green-light.

**Pull request closed without merging**